src/Resources/contao/controllers/BackendIndex.php

```diff
@@ -32,35 +45,12 @@ class BackendIndex extends \Backend
 	 */
 	public function __construct()
 	{
+		$this->container = System::getContainer();
+		$this->flashBag = $this->container->get('session')->getFlashBag();
+
 		$this->import('BackendUser', 'User');
 		parent::__construct();
 
-		// Login
-		if ($this->User->login())
-		{
-			$strUrl = 'contao/main.php';
-
-			// Redirect to the last page visited
-			if (\Input::get('referer', true) != '')
-			{
-				$strUrl = base64_decode(\Input::get('referer', true));
```

Comment:
The problem with the original symfony approach is, that it is stored in the session and not as URL parameter.
There might arise issues from this when opening a browser with multiple tabs.
The stored value in the session will be the one for the last tab that get's redirected to the login form then and cause all tabs to contain the same backend route after login.
We could implement it as stated here http://symfony.com/doc/current/security/form_login.html#control-the-redirect-url-from-inside-the-form
We would need to make our value in the template variable though as someone might reconfigure the application to use another parameter name (see config.yml in above link).